File: src/main/webui/src/app/business/oracle/files/boundary/files.resource.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { LogFile } from '../entity/log-file';
import { OracleFile } from '../entity/oracle-file';


@Injectable()
export class FilesResource {

    constructor(private http: HttpClient) { }

    files(): Observable<OracleFile[]> {
        return this.http.get<OracleFile[]>('files');
    }

	logFiles(): Observable<LogFile[]> {
        return this.http.get<LogFile[]>('files/log');
    }
}

File: src/main/java/pl/as/qora/oracle/files/boundary/FilesResource.java
package pl.as.qora.oracle.files.boundary;

import java.util.Collection;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;

import pl.as.qora.oracle.files.control.FilesService;
import pl.as.qora.oracle.files.entity.LogFile;
import pl.as.qora.oracle.files.entity.OracleFile;

@Path("files")
public class FilesResource {

    @Inject
    FilesService filesService;

    @GET
    public Collection<OracleFile> files() {
        return filesService.files();
    }
    
    @GET
    @Path("log")
    public Collection<LogFile> logFiles() {
        return filesService.logFiles();
    }
}
